### ISSUE TITLE
 Intercept package spec fails to connect when using gke-gcloud-auth-plugin 

### ISSUE THREAD

**Issue by Jell** [telepresenceio/telepresence#3065]
Given this spec:

```yaml
workloads:
  - name: echo
    namespace: jl-devops-test-app
    intercepts:
      - handler: echo-local
connection:
  context: "gke_treyd-development_europe-north1_treyd-development"
handlers:
  - name: echo-local
    docker:
      image: hashicorp/http-echo:latest
```

and the following kubeconfig:
```yaml
apiVersion: v1
clusters:
  - cluster: [REDACTED]
    name: gke_treyd-development_europe-north1_treyd-development
contexts:
- context:
    cluster: gke_treyd-development_europe-north1_treyd-development
    user: gke_treyd-development_europe-north1_treyd-development
  name: gke_treyd-development_europe-north1_treyd-development
current-context: gke_treyd-development_europe-north1_treyd-development
kind: Config
preferences: {}
users:
- name: gke_treyd-development_europe-north1_treyd-development
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      command: gke-gcloud-auth-plugin
      installHint: Install gke-gcloud-auth-plugin for use with kubectl by following
        https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
      provideClusterInfo: true
```

When I run:
```
telepresence intercept run my-intercept.yaml
```

I expect the connection to my GKE cluster to work, but instead I get:

```
telepresence connect: error: connector.Connect: initial cluster check failed: Get "https://1.2.3.4/version": getting credentials: exec: executable gke-gcloud-auth-plugin not found

It looks like you are trying to use a client-go credential plugin that is not installed.

To learn more about this feature, consult the documentation available at:
      https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins

Install gke-gcloud-auth-plugin for use with kubectl by following https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
telepresence intercept run: error: exit status 1
```

Note that if I run:

```
telepresence intercept --namespace jl-devops-test-app echo
```

without using an intercept package, the intercept works correctly. Kubectl works correctly. `gke-gcloud-auth-plugin` is in my path. Having a full path to `gke-gcloud-auth-plugin` in the kubeconfig returns the same error.

I'm using the following versions:
```
⎪▸ telepresence version
Client         : v2.11.1
Root Daemon    : v2.11.1
User Daemon    : v2.11.1
Traffic Manager: v2.11.1
```

**Comment by thallgren** [comment]
This will be fixed when #3055 is released.

**Comment by knlambert** [comment]
This issue was fixed in 2.12.
